master/buildbot/process/builder.py

```diff
@@ -302,12 +302,13 @@ def _startBuildFor(self, slavebuilder, buildrequests):
         cleanups = []
 
         def run_cleanups():
-            try:
-                while cleanups:
-                    fn = cleanups.pop()
-                    fn()
-            except:
-                log.err(failure.Failure(), "while running %r" % (run_cleanups,))
+            while cleanups:
+                fn = cleanups.pop()
+                try:
+                    if fn is not None:
+                        fn()
```

Comment:
If we were going to do that, the try/except should really only be around fn().  It shouldn't include the "if fn is not None" line.